vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -453,13 +461,32 @@ def _nixl_handshake(self, host: str, port: int):
             got_metadata_time = time.perf_counter()
 
             # Register Remote agent.
-            self.add_remote_agent(metadata)
+            self.add_remote_agent(metadata, rank)
```

Comment:
I think we should call this `self.add_remote_agent` separately from `handshake`

Since we always handshake with `rank 0`, we are calling `add_remote_agent()` more than once

Reply:
yes but there's no duplicate work done inside add_remote_agent apart from the nixl.add_remote_agent call (I assumed it would be convenient to keep a pointer to rank0). Local rank0 will only do that one call.  
I can work on clarity here.

Reply:
oh, because we skip if we already have it. I see. Ignore my comment.